File: static/js/registration.js
function toggleFields() {
    const customerFields = document.querySelectorAll('.customer-field');
    const companyFields = document.querySelectorAll('.company-field');
    const isCustomer = document.getElementById('customer').checked;

    customerFields.forEach(field => {
        field.style.display = isCustomer || !isCustomer ? 'block' : 'none';
    });
    companyFields.forEach(field => {
        field.style.display = !isCustomer ? 'flex' : 'none';
    });
}
function submitForm(event) {
    event.preventDefault();
    const formData = {
        companyName: document.getElementById('companyName').value,
        companyTelephone: document.getElementById('companyTelephone').value,
        firstName: document.getElementById('firstName').value,
        middleName: document.getElementById('middleName').value,
        lastName: document.getElementById('lastName').value,
        mobileNumber: document.getElementById('mobileNumber').value,
        emailId: document.getElementById('emailId').value,
        password: document.getElementById('password').value,
        state: document.getElementById('state').value,
        city: document.getElementById('city').value,
        street1: document.getElementById('street1').value,
        street2: document.getElementById('street2').value,
        pinCode: document.getElementById('pinCode').value,
        sex: document.getElementById('sex').value,
        age: document.getElementById('age').value,
        dob: document.getElementById('dob').value
    };

    fetch('/registration', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
    })
    .then(response => response.json())
    .then(data => {
        if (data.redirect_url) {
            window.location.href = data.redirect_url;
            console.log(window.location.href);
        } else {
            console.error('Error:', data);
        }
    })
    .catch((error) => {
        console.error('Error:', error);
    });
}

File: registration.py
from flask import Blueprint, request, jsonify, render_template, redirect, url_for, flash
from db_conn import db_connect
from werkzeug.security import generate_password_hash

registration_bp = Blueprint('registration', __name__)

# this is main index function
@registration_bp.route('/')
def index():
    return render_template('Registration.html')

@registration_bp.route('/registration', methods=['GET', 'POST'])
def register():
    if request.method == 'POST':
        data = request.json
        connection = None
        cursor = None

        try:
            connection = db_connect()
            cursor = connection.cursor()

            hashed_password = generate_password_hash(data['password'], method='pbkdf2:sha256')

            add_user = (
                "INSERT INTO User (companyName, companyTelephone, first_name, middle_name, last_name, mobile_number, email_id, password, state, city, street1, street2, pin_code, sex, Age, DOB) "
                "VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"
            )

            user_data = (
                data.get('companyName'), data.get('companyTelephone'), data['firstName'], data.get('middleName'), data['lastName'],
                data['mobileNumber'], data['emailId'], hashed_password, data['state'], data['city'], data['street1'],
                data.get('street2'), data['pinCode'], data['sex'], data['age'], data['dob']
            )

            cursor.execute(add_user, user_data)
            connection.commit()

            # return jsonify(message="User successfully created!")
            return jsonify({"message": "User successfully created!", "redirect_url": url_for('login.login')})
        except Error as e:
            return jsonify({"message": str(e)}), 500
        finally:
            if cursor:
                cursor.close()
            if connection:
                connection.close()
    else:
        return render_template('Registration.html')